addons/arsenal/functions/fnc_fillLeftPanel.sqf

```diff
@@ -0,0 +1,206 @@
+/*
+ * Author: Alganthe
+ * Fill  left panel.
+ *
+ * Arguments:
+ * 0: Arsenal display <DISPLAY>
+ * 1: Tab control <CONTROL>
+ *
+ * Return Value:
+ * None
+ *
+ * Public: No
+*/
+#include "script_component.hpp"
+#include "..\defines.hpp"
+
+params ["_display", "_control"];
+
+private _ctrlIDC = ctrlIDC _control;
+
+if !(isNil QGVAR(currentLeftPanel)) then {
+    private _previousCtrlBackground  = _display displayCtrl (GVAR(currentLeftPanel) - 1);
+    _previousCtrlBackground ctrlSetFade 1;
+    _previousCtrlBackground ctrlCommit FADE_DELAY;
+};
+
+private _ctrlBackground = _display displayCtrl (_ctrlIDC - 1);
+private _ctrlPanel = _display displayCtrl IDC_leftTabContent;
+_ctrlBackground ctrlSetFade 0;
+_ctrlBackground ctrlCommit FADE_DELAY;
+
+_ctrlPanel lbSetCurSel -1;
+
+// Handle icons and filling
+switch true do {
+    case (_ctrlIDC in [IDC_buttonPrimaryWeapon, IDC_buttonHandgun, IDC_buttonSecondaryWeapon]) : {
+        // Purge old data
+        lbClear _ctrlPanel;
+        private _addEmpty = _ctrlPanel lbadd format [" <%1>",localize "str_empty"];
+        _ctrlPanel lbsetvalue [_addEmpty, -1];
+
+        // Filling
+        switch (_ctrlIDC) do {
+            case IDC_buttonPrimaryWeapon : {
+                {
+                    ["CfgWeapons", _x, _ctrlPanel] call FUNC(addListBoxItem);
+                } foreach ((GVAR(virtualItems) select 0) select 0);
+            };
+
+            case IDC_buttonHandgun : {
+                {
+                    ["CfgWeapons", _x, _ctrlPanel] call FUNC(addListBoxItem);
+                } foreach ((GVAR(virtualItems) select 0) select 2);
+            };
+
+            case IDC_buttonSecondaryWeapon : {
+                {
+                    ["CfgWeapons", _x, _ctrlPanel] call FUNC(addListBoxItem);
+                } foreach ((GVAR(virtualItems) select 0) select 1);
+            };
+        };
+    };
+
+    case (_ctrlIDC in [IDC_buttonUniform, IDC_buttonVest, IDC_buttonBackpack]) : {
+
+        lbClear _ctrlPanel;
+        private _addEmpty = _ctrlPanel lbadd format [" <%1>",localize "str_empty"];
+        _ctrlPanel lbsetvalue [_addEmpty, -1];
+
+        // Filling
+        switch (_ctrlIDC) do {
```

Comment:
Again.
```
_index = 4+ ([IDC_buttonUniform,IDC_buttonVest,IDC_buttonBackpack] find _ctrlIDC);
```

Reply:
Third one is CfgVehicles, i'll keep this one.